plugins/out_cloudwatch_logs/cloudwatch_logs.c

```diff
@@ -528,6 +534,13 @@ static struct flb_config_map config_map[] = {
      " be created)"
     },
 
+    {
+     FLB_CONFIG_MAP_INT, "log_retention_days", "0",
+     0, FLB_FALSE, 0,
+     "If set to a number greater than zero, and newly create log group's "
+     "retention policy is set to this many days."
```

Comment:
Nit: For some reason that I do not understand the CloudWatch Logs API only supports a few values for retention:

```
[2020/11/16 05:56:10] [error] [output:cloudwatch_logs:cloudwatch_logs.0] PutRetentionPolicy API responded with error='InvalidParameterException', message='Invalid retention value. Valid values are: [1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]'
```

You could put the valid values list in the help text here